### PR TITLE
Refactor CheckpointMetadata to LastCheckpointInfo class

### PR DESCRIPTION
**Pull request by prakharjain09**
## Description

Rename CheckpointMetadata class to LastCheckpointInfo since that's what it actually is.

## How was this patch tested?

Existing UTs

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala [hunk 1]
```diff
@@ -54,246 +47,34 @@ import org.apache.spark.sql.types.StructType
...
- * @param size the number of actions in the checkpoint, -1 if the information is unavailable.
- * @param parts the number of parts when the checkpoint has multiple parts. None if this is a
- *              singular checkpoint
- * @param sizeInBytes the number of bytes of the checkpoint
- * @param numOfAddFiles the number of AddFile actions in the checkpoint
- * @param checkpointSchema the schema of the underlying checkpoint files
- * @param checksum the checksum of the [[CheckpointMetaData]].
- */
-@JsonPropertyOrder(Array("version", "size", "parts"))
-case class CheckpointMetaData(
```
Comment (ryan-johnson-databricks) [inline core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala:84]: Class moved to its own file, right? 
Any functional changes?

File: core/src/test/scala/org/apache/spark/sql/delta/CheckpointInstanceSuite.scala [hunk 2]
```diff
@@ -16,14 +16,13 @@
...
 
-import org.apache.spark.sql.delta.CheckpointMetaData.Format
+import org.apache.spark.sql.delta.CheckpointInstance.Format
 
 import org.apache.spark.SparkFunSuite
-import org.apache.spark.sql.test.SharedSparkSession
 
 class CheckpointInstanceSuite extends SparkFunSuite {
 
-  test("hello") {
```
Comment (ryan-johnson-databricks) [inline core/src/test/scala/org/apache/spark/sql/delta/CheckpointInstanceSuite.scala:26]: aside: quite the name there... thanks for fixing

File: core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala [hunk 3]
```diff
@@ -71,15 +71,15 @@ class EvolvabilitySuite extends EvolvabilitySuiteBase with DeltaSQLCommandTest {
     assert(contents.exists(_.contains(""""part":null""")), "null value should be written in json")
   }
 
-  testQuietly("parse old version CheckpointMetaData") {
-    assert(JsonUtils.mapper.readValue[CheckpointMetaData]("""{"version":1,"size":1}""")
-      === CheckpointMetaData(1, 1, None, None, None, None))
+  testQuietly("parse old version LastCheckpointInfoSuite") {
```
Comment (ryan-johnson-databricks) [inline core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala:74]:
```suggestion
  testQuietly("parse old version LastCheckpointInfo") {
```

File: core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala [hunk 4]
```diff
@@ -71,15 +71,15 @@ class EvolvabilitySuite extends EvolvabilitySuiteBase with DeltaSQLCommandTest {
...
+    assert(JsonUtils.mapper.readValue[LastCheckpointInfo]("""{"version":1,"size":1}""")
+      === LastCheckpointInfo(1, 1, None, None, None, None))
   }
 
-  test("parse partial version CheckpointMetaData") {
-    assert(JsonUtils.mapper.readValue[CheckpointMetaData](
+  test("parse partial version LastCheckpointInfoSuite") {
```
Comment (ryan-johnson-databricks) [inline core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala:79]:
```suggestion
  test("parse partial version LastCheckpointInfo") {
```

File: core/src/test/scala/org/apache/spark/sql/delta/CheckpointMetadataSuite.scala
Comment (ryan-johnson-databricks) [inline core/src/test/scala/org/apache/spark/sql/delta/CheckpointMetadataSuite.scala:1]: Shouldn't we rename the file as well?

File: core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala (hunk 1 above)
Comment (prakharjain09) [reply core/src/main/scala/org/apache/spark/sql/delta/Checkpoints.scala:84]: No functional changes.

File: core/src/test/scala/org/apache/spark/sql/delta/CheckpointMetadataSuite.scala
Comment (prakharjain09) [reply core/src/test/scala/org/apache/spark/sql/delta/CheckpointMetadataSuite.scala:1]: done.

File: core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala (hunk 3 above)
Comment (prakharjain09) [reply core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala]: fixed.

File: core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala (hunk 4 above)
Comment (prakharjain09) [reply core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuite.scala]: fixed.

File: core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuiteBase.scala [hunk 5]
```diff
@@ -46,7 +46,7 @@ abstract class EvolvabilitySuiteBase extends QueryTest with SharedSparkSession
     assert(deltaLog.snapshot.metadata.schema === StructType.fromDDL("id INT, value STRING"))
     assert(deltaLog.snapshot.metadata.partitionSchema === StructType.fromDDL("id INT"))
 
-    // Check we can load CheckpointMetaData
+    // Check we can load LastCheckpointInfoSuite
```
[the remark this answers is not in the document]
Comment (prakharjain09) [reply core/src/test/scala/org/apache/spark/sql/delta/EvolvabilitySuiteBase.scala]: fixed.

**Review by ryan-johnson-databricks — APPROVED** [review]
Looks good. New naming is much less confusing.

**Pull request closed without merging**